File: src/main/java/yunas/http/HttpMethod.java
package yunas.http;

/**
 * HttpMethod.
 *
 */
public enum HttpMethod  {

    GET("GET"),
    POST("POST"),
    PUT("PUT"),
    DELETE("DELETE"),
    HEAD("HEAD"),
    TRACE("TRACE"),
    OPTIONS("OPTIONS");

    private String value;

    HttpMethod(String value) {
        this.value = value;
    }
}

File: src/main/java/yunas/http/HttpMethod.kt
package yunas.http

/**
 * HttpMethod.
 *
 */
enum class HttpMethod private constructor(private val value: String) {

    GET("GET"),
    POST("POST"),
    PUT("PUT"),
    DELETE("DELETE"),
    HEAD("HEAD"),
    TRACE("TRACE"),
    OPTIONS("OPTIONS")
}
